src/libraries/Microsoft.PowerFx.Core/Public/Types/KnownRecordType.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Types
+{
+    public sealed class KnownRecordType : RecordType
```

Comment:
>KnownRecordType : RecordType [](http://example.com/codeflow?start=24&length=28)

Shouldn't we use RecordType : InternalRecordType instead (and rename existing RecordType) to keep the external type the same as the current one?

Reply:
Same idea for TableType

Reply:
Could you also add comments?